client/blocks/comment-detail/style.scss

```diff
@@ -383,11 +383,67 @@ a.comment-detail__author-more-element {
 }
 
 .comment-detail__reply {
-	line-height: 0;
+	// The inner working of these styles is covered here:
+	// http://alistapart.com/article/expanding-text-areas-made-elegant
+	.expanding-area {
+		min-height: 33px;
+		position: relative;
+		transition: min-height 0.2s ease-in-out;
+
+		&.focused {
+			min-height: 70px;
+		}
 
-	textarea {
-		border: none;
-		resize: vertical;
+		pre,
+		textarea {
+			font-family: $serif;
+			font-size: 14px;
+			line-height: 21px;
+			margin: 0;
+			max-height: 400px;
+			min-height: 33px;
+			overflow: hidden;
+			padding: 5px 60px 5px 5px;
+			white-space: pre-wrap;
+			word-wrap: break-word;
+		}
+
+		pre {
+			box-sizing: border-box;
+			display: block;
+			visibility: hidden;
+		}
+
+		textarea {
+			border: none;
+			height: 100%;
+			position: absolute;
+				left: 0;
+				top: 0;
+			resize: none;
+		}
+	}
+
+	.comment-detail__reply-submit {
+		color: lighten( $gray, 10% );
+		font-size: 12px;
+		font-weight: 600;
+		opacity: 0;
```

Comment:
I've used `opacity` for the transition that's subtle enough to make the React transition component way overkill.
Though, it being so little, I could drop id altogether and replace it with `display`.

Spot the differences (cc @drw158):

![jun-26-2017 17-31-02](https://user-images.githubusercontent.com/2070010/27549925-a06e103a-5a95-11e7-8a12-b15b8302f200.gif)
![jun-26-2017 17-33-17](https://user-images.githubusercontent.com/2070010/27549926-a06e5ff4-5a95-11e7-990a-62da599c4fe1.gif)